flow/project.py

```diff
@@ -1439,7 +1439,7 @@ def _fork(self, operation, timeout=None):
             fork(cmd=operation.cmd, timeout=timeout)
 
     def run(self, jobs=None, names=None, pretend=False, np=None, timeout=None, num=None,
-            num_passes=1, progress=False, shuffle=False):
+            num_passes=1, progress=False, order=None):
```

Comment:
You say in a comment that job ordering is done by default. Unless that changes, I think the default value should be `'by_job'`. Strings are not mutable in python so this would be a fine default value, and would more clear specify to users in the documentation that default ordering is `by-job`.

Reply:
Making the default value 'by-job' instead of `None` means that we are guaranteeing a specific order. If ever wanted to change that, then we need to go through a whole deprecation cycle. It's much safer to make no guarantees about the order by default to avoid future incompatibilities. The fact that `none` and `by-job` happen to be the same one for our specific implementation is therefore essentially of no relevance with respect to the default value to this argument.

There is one more reason why I'd prefer to have `None` be the default: Our execution model is generally based on the idea of a state-machine, where the definition order and execution order of operations should in principle not matter. The only reason I am introducing this option in the first place is to allow to speed up the execution under specific circumstances.

Reply:
That makes sense. I just didn't understand the reasoning before.